### DISCUSSION TITLE
SetUp AppWrite in expo Reactnative with using expresjs and mongoDb

### DISCUSSION THREAD

**Discussion by sudarshan20107049** [community/community#144753] [Programming Help]
### Body

"Hello community! I am a full stack developer who is new to React Native. I want to create an app for the Indian farmers community using React Native with Expo, Express.js server, and MongoDB database. I plan to implement login using Appwrite email OTP authentication. Could someone guide me through the basic setup process?"

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by richardfariax** [comment]
For Appwrite + Expo + Express, I'd keep auth and business data separate:

**Appwrite** handles email OTP login. In Expo, `createEmailToken` → user enters OTP → `createSession`. Store the session with `expo-secure-store` if you need persistence.

**Express + Mongo** holds farmer profiles, posts, etc. Don't duplicate user tables — pass the Appwrite session/JWT to your API and verify it server-side on each request.

**Common mistake:** logging in through Appwrite but calling Express routes without sending the session/JWT, so the backend has no idea who's logged in.

If you share whether you need offline support, the sync strategy changes a bit — but for a first version, online-only with JWT verification is enough.

**Comment by ketpatil77** [comment]
For this stack, I would separate responsibilities clearly:

- **Expo React Native app** = UI
- **Appwrite** = authentication (email OTP)
- **Express + MongoDB** = your domain data / business logic

A simple setup path:
1. Create Appwrite project
2. Enable email OTP auth
3. In Expo, use Appwrite SDK to start OTP login and create session
4. After login, call your Express API with the Appwrite session/JWT
5. In Express, verify the Appwrite user/session before reading or writing Mongo data

Important design choice:
- do **not** build a second custom auth system in Express if Appwrite already handles login
- let Appwrite prove identity
- let Express handle farmer profiles, crops, posts, marketplace data, etc.

For Expo specifically, I would start with:
- `expo-secure-store` for session/token persistence
- Appwrite SDK for auth
- Axios/fetch wrapper that always sends auth to Express
- one `/me` endpoint in Express to confirm end-to-end auth is working

Get OTP login working first, then protected API calls, then Mongo models. That order saves a lot of confusion.
